sources/packages/api/src/services/application/application.service.ts

```diff
@@ -28,45 +32,210 @@ export class ApplicationService extends RecordDataModelService<Application> {
   constructor(
     @Inject("Connection") connection: Connection,
     private readonly sequenceService: SequenceControlService,
+    private readonly fileService: StudentFileService,
   ) {
     super(connection.getRepository(Application));
     this.logger.log("[Created]");
   }
 
-  async createApplication(
+  /**
+   * Submits a Student Application.
+   * The system ensures that there is an application draft prior
+   * to the application submission.
+   * @param applicationId application id that must be updated to submitted.
+   * @param studentId student id for authorization validations.
+   * @param programYearId program year associated with the submission.
+   * @param applicationData dynamic data received from Form.IO form.
+   * @param associatedFiles associated uploaded files.
+   * @returns the updated application.
+   */
+  async submitApplication(
+    applicationId: number,
     studentId: number,
     programYearId: number,
-    applicationDto: CreateApplicationDto,
-    studentFiles: StudentFile[],
+    applicationData: any,
+    associatedFiles: string[],
   ): Promise<Application> {
-    // Create the new application.
-    const newApplication = new Application();
+    let application = await this.getApplicationToSave(
+      studentId,
+      ApplicationStatus.draft,
+      applicationId,
+    );
+    // If the application was not found it is because it does not belongs to the
+    // student or it is not in draft state. Either way it will be invalid.
+    if (!application) {
+      throw new Error(
+        "Student Application not found or it is not in the correct status to be submitted.",
+      );
+    }
+    // If the the draft was created under a different program year than
+    // the one being used to submit it, it is not valid.
+    if (application.programYear.id !== programYearId) {
+      throw new Error(
+        "Student Application program year is not the expect one.",
```

Comment:
Please change it to "Student Application program year is not the expected one"

Reply:
Done